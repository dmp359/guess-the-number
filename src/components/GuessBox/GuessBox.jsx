import React, { Component } from 'react';
import PropTypes from 'prop-types';
import LabelBox from 'components/LabelBox/LabelBox.jsx';

/*
* A simple React form to handle user input
*/
class GuessBox extends Component {
  constructor(props) {
    super(props);
    this.state = { value: '' };
    this.handleChange = this.handleChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  // Handle typing in input box. Sets state with value
  handleChange (e) {
    //  Error checking before parsing int should go here. Omitting for time.
    this.setState({ value: parseInt(e.target.value, 10) });
  }

  // Handle submitting of form
  onSubmit (e) {
    e.preventDefault(); // Don't refresh page
    this.props.onSubmit(this.state.value);
  }

  render() {
    return (
      <form className='form' onSubmit={this.onSubmit}>
        <LabelBox label="Guess: " onChange={this.handleChange} />
        <input type="submit" value="Submit" />
      </form>
    );
  }
}

GuessBox.propTypes = {
  onSubmit: PropTypes.func.isRequired,
}

export default GuessBox;
