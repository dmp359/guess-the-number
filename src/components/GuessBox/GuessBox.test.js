import React from 'react';
import { shallow } from 'enzyme';
import GuessBox from './GuessBox';

it('renders without crashing', () => {
  const component = shallow(<GuessBox onSubmit={() => {}}/>);
  expect(component.debug()).toMatchSnapshot();
});

it('handles parsing value of number', () => {
  const component = shallow(<GuessBox onSubmit={() => {}} />);
  expect(component.state('value')).toEqual('');
  component.instance().handleChange({ target: { value: '5' }});
  expect(component.state('value')).toEqual(5);
  component.instance().handleChange({ target: { value: 5 }});
  expect(component.state('value')).toEqual(5);
});
