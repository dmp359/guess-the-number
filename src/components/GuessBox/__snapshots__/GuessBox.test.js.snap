exports[`test renders without crashing 1`] = `
"<form className=\"form\" onSubmit={[Function: bound onSubmit]}>
  <LabelBox label=\"Guess: \" onChange={[Function: bound handleChange]} />
  <input type=\"submit\" value=\"Submit\" />
</form>"
`;
